From 43689e798fc791baee4e33392a9c4327f4d802fc Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 25 Feb 2025 15:52:20 +0100
Subject: [PATCH 1606/1695] plat-stm32mp1: add DDRMCE config in context V3

The DDRMCE regions configuration is saved in backup RAM context by TF-A
BL2 for standby mode in context V3 for STM32MP13 SoC, so add this element
to align the structure pm_mailbox with the struct backup_data_s define in
TF-A file plat/st/stm32mp1/stm32mp1_context.c.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b6d1e815cf6a1621e839c8e7e07fc5c26d09d062 plat-stm32mp1: add DDRMCE config in context V3]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ic503b6fea3ef3b37e47f2b2de9693a0c2ae8b87e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/439888
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/context.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/pm/context.c b/core/arch/arm/plat-stm32mp1/pm/context.c
index e77437324..50715cb0e 100644
--- a/core/arch/arm/plat-stm32mp1/pm/context.c
+++ b/core/arch/arm/plat-stm32mp1/pm/context.c
@@ -63,6 +63,15 @@
 
 #if defined(CFG_STM32MP13)
 #define MCE_KEY_SIZE_IN_BYTES		16
+
+/* IP configuration */
+#define MCE_IP_MAX_REGION_NB		1
+
+struct stm32_mce_region_s {
+	uint32_t encrypt_mode;	/* Specifies the region encryption mode */
+	uint32_t start_address;	/* Specifies the region start address */
+	uint32_t end_address;	/* Specifies the region end address */
+};
 #endif
 
 /*
@@ -116,6 +125,7 @@ struct pm_mailbox {
 	uint32_t bl2_end;
 #ifdef CFG_STM32MP13
 	uint8_t mce_mkey[MCE_KEY_SIZE_IN_BYTES];
+	struct stm32_mce_region_s mce_regions[MCE_IP_MAX_REGION_NB];
 #endif
 /* VERSION 4 */
 	uint32_t size;
-- 
2.47.2

